src/libponyc/pass/syntax.c

```diff
@@ -199,6 +199,7 @@ static bool check_permission(pass_opt_t* opt, const permission_def_t* def,
   int element, ast_t* actual, const char* context, ast_t* report_at)
 {
   pony_assert(def != NULL);
+  pony_assert(def->permissions != NULL);
```

Comment:
if we are going to check that permissions isn't NULL as an assert, we should do the same for def->desc